mrp_production_batch/models/mrp_production_batch.py

```diff
@@ -0,0 +1,226 @@
+                        .search([("lot_id", "=", lot)])
+                        .filtered(
+                            lambda x: x.move_id.raw_material_production_id
+                            in rec.production_ids
+                        )
+                        .move_id
+                    )
+                    raise ValidationError(
+                        _(
+                            "The following productions are using the same serial number in"
+                            " some of their components: %s"
+                            % move.mapped("raw_material_production_id.display_name")
+                        )
+                    )
+
+    def action_done(self):
+        self.ensure_one()
+        self.action_check()
+        self._check_unique_serial_lot_in_batch()
+        for production in self.with_context(
+            mrp_production_batch_create=True
+        ).production_ids:
+            production.action_generate_batch_serial()
+            production.button_mark_done()
+
+    def _get_common_action_view_production(self):
+        tree_view = self.env.ref(
+            "mrp_production_batch.mrp_production_view_tree_inherit"
+        )
+        form_view = self.env.ref(
+            "mrp_production_batch.mrp_production_view_form_inherit"
+        )
+        return {
+            "name": _("Detailed Operations"),
+            "type": "ir.actions.act_window",
+            "view_mode": "tree,form",
+            "res_model": "mrp.production",
+            "views": [(tree_view.id, "tree"), (form_view.id, "form")],
+            "view_id": tree_view.id,
+        }
+
+    def action_view_total_production(self):
+        self.ensure_one()
+        action = self._get_common_action_view_production()
+        action["domain"] = [("id", "in", self.production_ids.ids)]
+        return action
+
+    def action_view_production_to_review(self):
+        self.ensure_one()
+        action = self._get_common_action_view_production()
+        action["domain"] = [("id", "in", self.to_review_production_ids.ids)]
+        return action
+
+    def action_view_production_ready(self):
+        self.ensure_one()
+        action = self._get_common_action_view_production()
+        action["domain"] = [("id", "in", self.ready_production_ids.ids)]
+        return action
+
+    @api.depends("name", "state")
+    def name_get(self):
+        result = []
+        for rec in self:
+            if rec.state == "draft":
+                state = _("Draft")
+                hour = fields.Datetime.from_string(rec.create_date).strftime("%H:%M")
+                date = fields.Datetime.from_string(rec.create_date).strftime("%d/%m/%Y")
+                name = "%s [%s %s] %s" % (
+                    state,
+                    hour,
+                    date,
+                    rec.operation_type.display_name,
+                )
+            else:
+                name = rec.name
+            result.append((rec.id, name))
+        return result
```

Comment:
This is not taking into account the timezone, fix it using the following example
https://github.com/nuobit/odoo-addons/blob/5dd2c6417de2157580550fa2c32f622acaa371d2/account_invoice_batches/models/account_invoice_batch.py#L90-L109